sandbox/modules/identity/src/main/java/org/opensearch/identity/User.java

```diff
@@ -34,12 +35,13 @@ public User() {}
      * @param attributes A map of custom attributes
      * @throws IllegalArgumentException if username or bcryptHash is null or empty
      */
-    public User(final String username, final String bcryptHash, Map<String, String> attributes) {
+    public User(final String username, final String bcryptHash, Map<String, String> attributes, List<String> permissions) {
```

Comment:
I am not sure if we are planning on storing the permissions as part of the User object. 

I don't have an issue with doing it this way but I had thought there may have been past discussion of just referencing a common database-type view of an index holding all the permissions. Do you have any concrete thoughts about why we should have an active permission list be an attribute of an user? 

How do you plan to store the permissions associated with non-User subjects? For example, I don't think we have the same type of objects for plugins and extensions but I believe we intend to be able to grant them permissions similar to users.

Reply:
I added them on the User to show an example of how permissions checks can work. I can add a TODO statement in the code to make it clear that this isn't the final place where permissions will reside.

Reply:
No worries then. I just did not want to have you write a whole thing with the expectation that they would be part of the User object when they probably will be stored elsewhere. As long as you know it makes sense to me.